cpp/dolfin/mesh/Topology.cpp

```diff
@@ -135,6 +135,54 @@ const std::vector<std::int32_t>& Topology::cell_owner() const
   return _cell_owner;
 }
 //-----------------------------------------------------------------------------
+std::vector<bool> Topology::surface_entity_marker(int dim) const
+{
+  const int tdim = this->dim();
+
+  if (dim >= tdim or dim < 0)
+  {
+    throw std::runtime_error("Invalid entity dimension: "
+                             + std::to_string(dim));
+  }
+
+  std::shared_ptr<const Connectivity> connectivity_facet_cell
+      = connectivity(tdim - 1, tdim);
+
+  std::vector<bool> marker(size(dim), false);
+
+  const int num_facets = size(tdim - 1);
+  // Special case for facets
+  if (dim == tdim - 1)
+  {
+    for (int i = 0; i < num_facets; ++i)
+    {
+      if (connectivity_facet_cell->size_global(i) == 1)
+        marker[i] = true;
+    }
+    return marker;
+  }
+
+  // Get connectivity from facet to entities of interest (vertices or edges)
+  std::shared_ptr<const Connectivity> connectivity_facet_entity
```

Comment:
This connectivity has already been fetched.

Reply:
It hasn't, but it looks very similar. Previously it was `(tdim - 1, tdim)` and nw it is `(tdim - 1, dim)`.